run.py

```diff
@@ -906,119 +905,93 @@ def hvreads(args):
-            "--no-unal",
-            "--threads", "24",
-            # Including an @SQ line makes the .sam files really big, and we
-            # don't actually need to know the lengths of all the genomes we
-            # considered aligning against. When we have a small number of reads
-            # per sample this can bloat the files 100x.  We'd like to do
-            # "--no-sq" bur pysam chokes on SAM files that are missing the @SQ
-            # header lines.
-            "-S", sam_out,
-         ]
-
-         assert bool(pair1) == bool(pair2)
-         if pair1:
-            cmd.extend(["-1", "pair1.fastq", "-2", "pair2.fastq"])
-         if combined:
-            cmd.extend(["-U", "combined.fastq"])
+            outputs.append(output)
 
-         subprocess.check_call(cmd)
+            # TODO(jefftk): do we have a problem when read1 or read2 is too
+            # short?  I remember Bowtie choking on these before.
+
+            full_inputs = [
+               "%s/%s/cleaned/%s" % (S3_BUCKET, args.bioproject, input_fname)
+               for input_fname in inputs]
 
-         with open(os.path.join(
-               THISDIR, "bowtie", "genomeid-to-taxid.json")) as inf:
-            genomeid_to_taxid = json.load(inf)
-
-         with pysam.AlignmentFile(sam_out, "r")  as sam:
-            with open(output, "w") as outf:
-               for record in sam:
-                  genomeid = sam.get_reference_name(record.reference_id)
-                  taxid, genome_name = genomeid_to_taxid[genomeid]
-                  outf.write(
-                     "%s\t%s\t%s\t%s\t%s\t%s\t%s\n" % (
-                        record.query_name,
-                        genomeid,
-                        taxid,
-                        record.cigarstring,
-                        record.reference_start,
-                        record.get_tag("AS"),
-                        len(record.query_sequence)))
+            subprocess.check_call([
+               os.path.join(THISDIR, "compute-alignments.sh"),
+               output,
+               *full_inputs,
+            ])
+
+         with gzip.open(combined_output_compressed, "wt") as outf:
+            for output in outputs:
+               with open(output) as inf:
+                  for line in inf:
+                     if line.startswith("@"): continue
+                     bits = line.rstrip("\n").split("\t")
+
+                     query_name = bits[0]
+                     genomeid = bits[2]
+                     ref_start = bits[3]
+                     cigarstring = bits[5]
+                     query_len = len(bits[9])
+
+                     as_val = None
+                     for token in bits[11:]:
+                        if token.startswith("AS:i:"):
+                           as_val = token.replace("AS:i:", "")
+                     assert as_val
+
+                     # The AS value is 1-indexed, but we use 0-indexing.
+                     query_pos = int(as_val) - 1
```

Comment:
How does the variable name `query_pos` match onto what it represents (the alignment score)?

Reply:
Good catch!  Not only did I call it the wrong thing, I adjusted the wrong number for 1-indexing.  Fixed in 14ec3fa